src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -549,7 +549,9 @@ def build_load_balancer_resource(cmd, name, location, tags, backend_pool_name, n
                     },
                     'protocol': 'tcp',
                     'frontendPortRangeStart': '50000',
-                    'frontendPortRangeEnd': '50119',
+                    # keep 50119 as minimum for backward compat, and ensure over-provision is taken care of
+                    'frontendPortRangeEnd': str(max(50119,
+                                                    49999 + instance_count * (1 if disable_overprovision else 2))),
```

Comment:
When we disable overprovisioning, does compute tend to provision twice the instances of VM's in the scaleset. Or are we selecting 2 here because it seems like a reasonable number.

Reply:
When over provision is enabled, compute might provision up to twice of the instance count, so yes 2 is a reasonable.